storage/skip_list/src/skip_list.rs

```diff
@@ -33,17 +33,17 @@ impl<C: ListValue> KeyValueSchema for SkipList<C> {
     }
 }
 
-impl<C: ListValue> SkipList<C> {
+impl<K: Codec, V: Codec, C: ListValue<K, V>> SkipList<K, V, C> {
     /// Create new list in given database
-    pub fn new(list_id: SkipListId, db: Arc<rocksdb::DB>) -> Result<Self, SkipListError> {
-        let list_db: Arc<SkipListDatabase<C>> = db.clone();
+    pub fn new(list_id: SkipListId, persistent_storage: &storage::persistent::PersistentStorage) -> Result<Self, SkipListError> {
```

Comment:
Do we need to pass the whole `PersistentStorage` here? This is a specialized type which should not have a need for the whole persistent storage.

Reply:
Fixed in c97f893